pkg/virt-controller/services/template.go

```diff
@@ -401,6 +405,28 @@ func getMemoryOverhead(domain v1.DomainSpec) *resource.Quantity {
 	return overhead
 }
 
+func getPortsFromVMI(vmi *v1.VirtualMachineInstance) []k8sv1.ContainerPort {
+	ports := make([]k8sv1.ContainerPort, 0)
+
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if iface.Ports != nil {
+			for _, port := range iface.Ports {
+				if port.Protocol == "" {
+					port.Protocol = "TCP"
+				}
+
+				ports = append(ports, k8sv1.ContainerPort{Protocol: k8sv1.Protocol(port.Protocol), Name: port.Name, ContainerPort: port.Port})
```

Comment:
If that was indeed the intent to allow duplicate ports in API, would it make sense to dedup them here to avoid passing duplicates to container spec? (I am still puzzled why we may want to accept duplicates in API so maybe this comment is moot.)